File: CPP/binarysearch.cpp
#include <bits/stdc++.h>
using namespace std;
bool bin_search(int a[],int n,int k)
{
    int lo=0,hi=n-1;
    while(lo<=hi)
    {
        int mid=(lo+hi)/2;
        if(a[mid]==k)
        return 1;
        if(a[mid]>k)
        hi=mid-1;
        else
        lo=mid+1;
    }
    return -1;
}
int main() {
	    int arr[] = { 1000, 11, 445,  
                  1, 330, 3000 }; 
    int arr_size = 6; 
    int k=330;
       if(bin_search(arr,arr_size,k))
       cout<<"Element Found";
       else
       cout<<"Element Not Found";
	return 0;
}

File: Python/Binary Search.py
# Returns index of x in arr if present, else -1 
def binary_search(arr, low, high, x): 

	# Check base case 
	if high >= low: 

		mid = (high + low) // 2

		# If element is present at the middle itself 
		if arr[mid] == x: 
			return mid 

		# If element is smaller than mid, then it can only 
		# be present in left subarray 
		elif arr[mid] > x: 
			return binary_search(arr, low, mid - 1, x) 

		# Else the element can only be present in right subarray 
		else: 
			return binary_search(arr, mid + 1, high, x) 

	else: 
		# Element is not present in the array 
		return -1

# Test array 
arr = input("Enter list ex: 1,2,3 \n").split(',') 
x = int(input("Enter Number to be Search"))

# Function call 
result = binary_search(arr, 0, len(arr)-1, x) 

if result != -1: 
	print("Element is present at index", str(result)) 
else: 
	print("Element is not present in array") 
